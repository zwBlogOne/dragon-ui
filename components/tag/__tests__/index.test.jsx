import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tag from '../index';

describe('Tag', () => {
  it('renders normal Tag correctly', () => {
    const wrapper = render(
      <div>
        <Tag>default</Tag>
        <Tag theme="info">info</Tag>
        <Tag theme="success">success</Tag>
        <Tag theme="warning">warning</Tag>
        <Tag theme="error">error</Tag>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders disabled Tag correctly', () => {
    const wrapper = render(
      <div>
        <Tag disabled>default</Tag>
        <Tag disabled theme="info">info</Tag>
        <Tag disabled theme="success">success</Tag>
        <Tag disabled theme="warning">warning</Tag>
        <Tag disabled theme="error">error</Tag>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Tag correctly with more props', () => {
    const wrapper = render(
      <div>
        <Tag radius>default</Tag>
        <Tag radius size="xl" theme="info">info</Tag>
        <Tag radius size="lg"theme="success">success</Tag>
        <Tag radius size="sm" theme="warning">warning</Tag>
        <Tag radius size="xs" theme="error">error</Tag>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when closing tag', () => {
    const onClose = jest.fn();
    const wrapper = mount(
      <Tag theme="info" onClose={onClose}>可关闭标签</Tag>
    );

    wrapper.find('.ui-icon').simulate('click');
    expect(onClose).toHaveBeenCalled();
  });
});

