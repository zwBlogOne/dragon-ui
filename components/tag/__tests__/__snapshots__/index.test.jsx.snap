// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders Tag correctly with more props 1`] = `
<div>
  <div
    class="ui-tag radius theme-default"
  >
    default
  </div>
  <div
    class="ui-tag radius theme-info size-xl"
  >
    info
  </div>
  <div
    class="ui-tag radius theme-success size-lg"
  >
    success
  </div>
  <div
    class="ui-tag radius theme-warning size-sm"
  >
    warning
  </div>
  <div
    class="ui-tag radius theme-error size-xs"
  >
    error
  </div>
</div>
`;

exports[`Tag renders disabled Tag correctly 1`] = `
<div>
  <div
    class="ui-tag disabled theme-default"
  >
    default
  </div>
  <div
    class="ui-tag disabled theme-info"
  >
    info
  </div>
  <div
    class="ui-tag disabled theme-success"
  >
    success
  </div>
  <div
    class="ui-tag disabled theme-warning"
  >
    warning
  </div>
  <div
    class="ui-tag disabled theme-error"
  >
    error
  </div>
</div>
`;

exports[`Tag renders normal Tag correctly 1`] = `
<div>
  <div
    class="ui-tag theme-default"
  >
    default
  </div>
  <div
    class="ui-tag theme-info"
  >
    info
  </div>
  <div
    class="ui-tag theme-success"
  >
    success
  </div>
  <div
    class="ui-tag theme-warning"
  >
    warning
  </div>
  <div
    class="ui-tag theme-error"
  >
    error
  </div>
</div>
`;
